jbmc/src/java_bytecode/remove_exceptions.cpp

```diff
@@ -587,6 +589,14 @@ void remove_exceptionst::instrument_exceptions(
     }
   }
 
+  // INITIALIZE_FUNCTION should not contain any exception handling branches for
+  // two reasons: (1) with symex-driven lazy loading it means that code that
+  // references @inflight_exception might be generated before
+  // @inflight_exception is initialized; (2) symex can analyze
+  // INITIALIZE_FUNCTION much faster if it doesn't contain any branching.
+  INVARIANT(function_identifier != INITIALIZE_FUNCTION || !did_something,
```

Comment:
Given that seemingly you had to modify the source code of a test to make this `INVARIANT` pass I am wondering whether this really is an invariant?

Reply:
What macro would capture my intent better? Quite a lot of work went into making sure that branches for exception-handling weren't created in `__CPROVER_initialize`. We went to make sure it doesn't creep back in by having an assertion which fails when it does.

Reply:
If user input can violate this constraint (which I thought was possible, given you had to change a test), then it should be an exception (for example ` unsupported_operation_exceptiont`).

Reply:
The "user input" in those tests is a cut-down model for java.lang.Class, implementing special CPROVER functions like `cproverNondetInitialize` or `cproverInitializeClassLiteral`. I don't think we expect users to be providing anything like that.

Reply:
Oh, I see, I should probably just remain quiet about any Java-related code :-)  Thanks for clarifying this, the invariant does seem to make a lot of sense in that case.